proto/src/rr/domain/label.rs

```diff
@@ -322,13 +334,29 @@ impl IntoLabel for Label {
 
 impl<'a> IntoLabel for &'a str {
```

Comment:
To reduce confusion in this change, we should not support direct str conversion, i.e. only from_ascii or from_bytes only.